### PR TITLE
 Fixes #26283 - hammer sync-plan update 

### PR DESCRIPTION
**Pull request by sjha4**
Steps to Reproduce:
1. hammer sync-plan create  --name BugTest --sync-date "2019/02/01" --enabled 1  --interval "custom cron" --cron-expression "1 2 3 4 5" --organization-id 1 
2. hammer sync-plan info --name BugTest --organization-id 1

3. hammer sync-plan update --name BugTest --organization-id 1 --description Hello
Could not update the sync plan:
  Cron expression is not valid!

4. hammer sync-plan update --name BugTest --organization-id 1 --description Hello --cron-expression "1 2 3 4 5"
Could not update the sync plan:
  Cron expression is not valid!

Expected result: These are all valid operations and should succeed.

### CODE REVIEW

**Comment by omkarkhatavkar** [comment]
Some Scenarios Tested : 
**Passed Sceanrios:**
```
[vagrant@centos7-hammer-devel ~]$ hammer sync-plan update --name BugTest1 --organization-id 2 --description Hello --enabled no --organization-id 2
Sync plan updated.
[vagrant@centos7-hammer-devel ~]$ hammer sync-plan update --name BugTest1 --organization-id 2 --description Hello --enabled yes --organization-id 2
Sync plan updated.
[vagrant@centos7-hammer-devel ~]$ hammer sync-plan update --name BugTest1 --organization-id 2 --description Hello --interval "custom cron" --cron-expression "SAdsadsad" --organization-id 2
Could not update the sync plan:
  Cron expression is not valid!
[vagrant@centos7-hammer-devel ~]$ hammer sync-plan update --name BugTest1 --organization-id 2 --description Hello --interval "daily" --cron-expression "1 2 3 4 5" --organization-id 2
Could not update the sync plan:
  Validation failed: Custom cron expression only needs to be set for interval value of custom cron
[vagrant@centos7-hammer-devel ~]$ 
[vagrant@centos7-hammer-devel ~]$ hammer sync-plan update --name BugTest1 --organization-id 2 --description '' --interval "custom cron" --organization-id 2
Could not update the sync plan:
  Cron expression is not valid! 
```

**Failed Scenario:** 
```
[vagrant@centos7-hammer-devel ~]$ hammer  sync-plan create --name BugTest_asdsadsa --sync-date "2019/02/01" --enabled 1 --interval "daily" --cron-expression "Asdsadsad" --organization-id 2
Sync plan created.
```
Currently while updating custom cron we adding validation, @sjha4 do you think we need to add validation while creating sync plan, if some one passes the custom cron. Otherwise looks good!

**Comment by sjha4** [comment]
@omkarkhatavkar : So when creating sync plan, if the interval is not custom cron, we overwrite the cron-expression to ''. Therefore, no validation is thrown. You will get the expected validation error when saving with interval custom cron.

**Comment by omkarkhatavkar** [comment]
@sjha4 i'm good with that. Just wanted to know your thoughts.

File: app/models/katello/sync_plan.rb [hunk 1]
```diff
@@ -56,7 +56,7 @@ def save_with_logic!
 
     def update_attributes_with_logics!(params)
       transaction do
-        params["cron_expression"] = '' unless params["interval"].eql? CUSTOM_CRON
+        params["cron_expression"] = '' if (params.key?("interval") && !params["interval"].eql?(CUSTOM_CRON) && self.interval.eql?(CUSTOM_CRON))
```
Comment (jlsherrill) [inline app/models/katello/sync_plan.rb]: can you add a test to test this scenario?

Comment (sjha4) [reply app/models/katello/sync_plan.rb]: Added some tests and also removed self.save! from update method as discussed offline.

**Pull request merged**